src/recipe/recipe.service.ts

```diff
@@ -35,35 +57,117 @@ export class RecipeService {
       throw new NotFoundException();
     }
 
+    if (
+      user.id != recipe.authorId &&
+      user.role != 'ADMIN' &&
+      !recipe.isPublic
+    ) {
+      throw new ForbiddenException();
+    }
+
+    if (user.id != recipe.authorId && !recipe.isPublic) {
+      throw new ForbiddenException();
+    }
+
     return recipe;
   }
 
-  fetchAllRecipes(): Promise<Recipe[]> {
-    return this.prisma.recipe.findMany();
+  async fetchUsersRecipes(email: string, userId: number): Promise<Recipe[]> {
+    const recipesOwner = await this.prisma.user.findUnique({
+      where: { email },
+    });
+    if (!recipesOwner) {
+      throw new NotFoundException();
+    }
+    const visitor = await this.prisma.user.findUnique({
+      where: { id: userId },
+    });
+
+    if (recipesOwner.id === visitor.id || visitor.role == 'ADMIN') {
+      return this.prisma.recipe.findMany({
+        where: {
+          authorId: recipesOwner.id,
+        },
+      });
+    } else {
+      return this.prisma.recipe.findMany({
+        where: {
+          authorId: recipesOwner.id,
+          isPublic: true,
+        },
+      });
+    }
+  }
```

Comment:
You should think about some way to avoid having that many if statements.

Service that is responsible for making calls to DB should not contain that kind of logic
`recipesOwner.id === visitor.id || visitor.role == 'ADMIN'` 
or:
`      user.id != recipe.authorId &&
      user.role != 'ADMIN' &&
      !recipe.isPublic`

this looks like a business logic that should be handled on a different level.

Reply:
This has been fixed with repository layer, which separates these things